Spatial-Indexer gives an IllegalAccessError

### Version

Jena Fuseki 6.1

### What happened?

The spatial indexer is configured in the assembler.  Data can be loaded into the triplestore and queried using geo functions. The web page for the spatial indexer shows correctly, and it identifies the default graphs, but when asked to index there is an error in the JS.  When using curl to run the indexation commands from a shell the following is an example of what is returned, irrespective of the command.


Running on windows with Azul Zulu Java 24

### Relevant output and stacktrace

#### Startup
```
20:32:48 INFO  Fuseki          :: Apache Jena Fuseki 6.1.0
20:32:48 INFO  Config          :: Fuseki Base = C:\apache-jena-fuseki-6.1.0_spatial\run
20:32:48 INFO  Config          :: Load configuration: file:///C:/apache-jena-fuseki-6.1.0_spatial/run/configuration/stress_test.ttl
20:32:49 INFO  GeoSPARQLOperations :: Applying GeoSPARQL Schema - Started
20:32:49 INFO  GeoSPARQLOperations :: GeoSPARQL schema applied to graph: default
20:32:49 INFO  GeoSPARQLOperations :: Applying GeoSPARQL Schema - Completed
20:32:49 INFO  SpatialIndexLib :: Building Spatial Index - Started
20:32:49 INFO  SpatialIndexLib :: Building Spatial Index - Completed
20:32:49 INFO  SpatialIndexIoKryo :: Saving Spatial Index - Started: C:\apache-jena-fuseki-6.1.0_spatial\run\databases\stress_test\spatial.index
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::arrayBaseOffset has been called by com.esotericsoftware.kryo.kryo5.unsafe.UnsafeUtil (file:/C:/apache-jena-fuseki-6.1.0_spatial/fuseki-server.jar)
WARNING: Please consider reporting this to the maintainers of class com.esotericsoftware.kryo.kryo5.unsafe.UnsafeUtil
WARNING: sun.misc.Unsafe::arrayBaseOffset will be removed in a future release
20:32:49 INFO  SpatialIndexIoKryo :: Saving Spatial Index - Success: C:\apache-jena-fuseki-6.1.0_spatial\run\databases\stress_test\spatial.index
20:32:49 INFO  Config          :: Database: /ds
20:32:49 INFO  Config          :: UI Base: fuseki-server.jar
20:32:50 INFO  Shiro           :: Shiro configuration: file:C:\apache-jena-fuseki-6.1.0_spatial\run\shiro.ini
20:32:50 INFO  Server          :: Path = /ds
20:32:50 INFO  Server          :: System
20:32:50 INFO  Server          ::   Memory: 4.0 GiB
20:32:50 INFO  Server          ::   PID:    24308
20:32:50 INFO  Server          ::   Port:   http=3030
20:32:50 INFO  Server          :: Start Fuseki
20:32:50 INFO  Fuseki          :: [2] GET http://localhost:3030/ds/spatial-indexer?command=events
20:32:50 INFO  Fuseki          :: [1] GET http://localhost:3030/ds/spatial-indexer?command=events
20:32:50 INFO  Fuseki          :: [1] -1 -1 (15 ms)
20:32:50 INFO  Fuseki          :: [2] -1 -1 (15 ms)
```

#### Error output
```shell

        ... 56 more
20:06:27 INFO  Fuseki          :: [9] 500 Server Error (2 ms)
20:07:00 INFO  Fuseki          :: [10] POST http://localhost:3030/ds/spatial-indexer?command=clean
20:07:00 ERROR Fuseki          :: An unexpected error occurred.
java.lang.IllegalAccessError: 'boolean org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib$1.isTerminated()'
        at org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib.lambda$scheduleOnceCleanTask$1(SpatialIndexLib.java:335)
        at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1956)
        at org.apache.jena.sparql.util.Context.compute(Context.java:399)
        at org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib.scheduleOnceCleanTask(SpatialIndexLib.java:333)
        at org.apache.jena.fuseki.mod.geosparql.SpatialIndexerService.doClean(SpatialIndexerService.java:393)
        at org.apache.jena.fuseki.mod.geosparql.SpatialIndexerService.doPost(SpatialIndexerService.java:239)
        at org.apache.jena.fuseki.servlets.ActionREST.doPost$(ActionREST.java:90)
        at org.apache.jena.fuseki.servlets.ActionREST.execute(ActionREST.java:45)
        at org.apache.jena.fuseki.servlets.ActionService.executeLifecycle(ActionService.java:61)
        at org.apache.jena.fuseki.servlets.ActionREST.process(ActionREST.java:155)
        at org.apache.jena.fuseki.servlets.ActionExecLib.execActionSub(ActionExecLib.java:127)
        at org.apache.jena.fuseki.servlets.ActionExecLib.execAction(ActionExecLib.java:101)
        at org.apache.jena.fuseki.server.Dispatcher.dispatchAction(Dispatcher.java:242)
        at org.apache.jena.fuseki.server.Dispatcher.process(Dispatcher.java:232)
        at org.apache.jena.fuseki.server.Dispatcher.dispatch(Dispatcher.java:150)
        at org.apache.jena.fuseki.servlets.FusekiFilter.doFilter(FusekiFilter.java:52)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:65)
        at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:109)
        at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:138)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:70)
        at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:109)
        at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:138)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:70)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:463)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.lambda$doFilterInternal$0(AbstractShiroFilter.java:378)
        at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:91)
        at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:84)
        at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:389)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:376)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.apache.jena.fuseki.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:351)
        at org.apache.jena.fuseki.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:304)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1602)
        at org.eclipse.jetty.ee11.servlet.ServletChannel.dispatch(ServletChannel.java:871)
        at org.eclipse.jetty.ee11.servlet.ServletChannel.handle(ServletChannel.java:449)
        at org.eclipse.jetty.ee11.servlet.ServletHandler.handle(ServletHandler.java:469)
        at org.eclipse.jetty.ee11.servlet.SessionHandler.handle(SessionHandler.java:763)
        at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1226)
        at org.eclipse.jetty.server.Server.handle(Server.java:197)
        at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:788)
        at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:420)
        at org.eclipse.jetty.server.internal.HttpConnection$FillableCallback.succeeded(HttpConnection.java:1809)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:54)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:492)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.epcRunTask(AdaptiveExecutionStrategy.java:428)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:401)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:255)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:204)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:317)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:1009)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1240)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1194)
        at java.base/java.lang.Thread.run(Thread.java:1447)
20:07:00 WARN  Fuseki          :: [10] ActionErrorException with cause
org.apache.jena.fuseki.servlets.ActionErrorException
        at org.apache.jena.fuseki.servlets.ServletOps.actionErrorException(ServletOps.java:277)
        at org.apache.jena.fuseki.servlets.ServletOps.errorOccurred(ServletOps.java:272)
        at org.apache.jena.fuseki.servlets.ServletOps.errorOccurred(ServletOps.java:266)
        at org.apache.jena.fuseki.mod.geosparql.SpatialIndexerService.doPost(SpatialIndexerService.java:248)
        at org.apache.jena.fuseki.servlets.ActionREST.doPost$(ActionREST.java:90)
        at org.apache.jena.fuseki.servlets.ActionREST.execute(ActionREST.java:45)
        at org.apache.jena.fuseki.servlets.ActionService.executeLifecycle(ActionService.java:61)
        at org.apache.jena.fuseki.servlets.ActionREST.process(ActionREST.java:155)
        at org.apache.jena.fuseki.servlets.ActionExecLib.execActionSub(ActionExecLib.java:127)
        at org.apache.jena.fuseki.servlets.ActionExecLib.execAction(ActionExecLib.java:101)
        at org.apache.jena.fuseki.server.Dispatcher.dispatchAction(Dispatcher.java:242)
        at org.apache.jena.fuseki.server.Dispatcher.process(Dispatcher.java:232)
        at org.apache.jena.fuseki.server.Dispatcher.dispatch(Dispatcher.java:150)
        at org.apache.jena.fuseki.servlets.FusekiFilter.doFilter(FusekiFilter.java:52)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:65)
        at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:109)
        at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:138)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:70)
        at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:109)
        at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:138)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:70)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:463)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.lambda$doFilterInternal$0(AbstractShiroFilter.java:378)
        at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:91)
        at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:84)
        at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:389)
        at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:376)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:156)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.apache.jena.fuseki.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:351)
        at org.apache.jena.fuseki.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:304)
        at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
        at org.eclipse.jetty.ee11.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1602)
        at org.eclipse.jetty.ee11.servlet.ServletChannel.dispatch(ServletChannel.java:871)
        at org.eclipse.jetty.ee11.servlet.ServletChannel.handle(ServletChannel.java:449)
        at org.eclipse.jetty.ee11.servlet.ServletHandler.handle(ServletHandler.java:469)
        at org.eclipse.jetty.ee11.servlet.SessionHandler.handle(SessionHandler.java:763)
        at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1226)
        at org.eclipse.jetty.server.Server.handle(Server.java:197)
        at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:788)
        at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:420)
        at org.eclipse.jetty.server.internal.HttpConnection$FillableCallback.succeeded(HttpConnection.java:1809)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:54)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:492)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.epcRunTask(AdaptiveExecutionStrategy.java:428)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:401)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:255)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:204)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:317)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:1009)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1240)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1194)
        at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: java.lang.IllegalAccessError: 'boolean org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib$1.isTerminated()'
        at org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib.lambda$scheduleOnceCleanTask$1(SpatialIndexLib.java:335)
        at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1956)
        at org.apache.jena.sparql.util.Context.compute(Context.java:399)
        at org.apache.jena.geosparql.spatial.index.v2.SpatialIndexLib.scheduleOnceCleanTask(SpatialIndexLib.java:333)
        at org.apache.jena.fuseki.mod.geosparql.SpatialIndexerService.doClean(SpatialIndexerService.java:393)
        at org.apache.jena.fuseki.mod.geosparql.SpatialIndexerService.doPost(SpatialIndexerService.java:239)
        ... 56 more
20:07:00 INFO  Fuseki          :: [10] 500 Server Error (2 ms)

```


### Are you interested in making a pull request?

None

--- a/jena-fuseki2/jena-fuseki-mod-geosparql/src/main/java/org/apache/jena/fuseki/mod/geosparql/SpatialIndexerService.java
+++ b/jena-fuseki2/jena-fuseki-mod-geosparql/src/main/java/org/apache/jena/fuseki/mod/geosparql/SpatialIndexerService.java
@@ -433,8 +433,8 @@ public class SpatialIndexerService extends BaseActionREST {
             status.addProperty("isIndexing", false);
             time = 0;
         } else {
-            status.addProperty("isIndexing", !task.isTerminated());
-            if (!task.isTerminated()) {
+            status.addProperty("isIndexing", !task.hasTerminated());
+            if (!task.hasTerminated()) {
                 status.addProperty("isAborting", task.isAborting());
                 time = !task.isAborting() ? task.getStartTime() : task.getAbortTime();
             } else {

--- a/jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/index/v2/SpatialIndexLib.java
+++ b/jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/index/v2/SpatialIndexLib.java
@@ -226,7 +226,7 @@ public class SpatialIndexLib {
 
         BasicTask task = cxt.compute(SpatialIndexConstants.symSpatialIndexTask, (key, priorTaskObj) -> {
             BasicTask priorTask = (BasicTask)priorTaskObj;
-            if (priorTask != null && !priorTask.isTerminated()) {
+            if (priorTask != null && !priorTask.hasTerminated()) {
                 throw new RuntimeException("A spatial indexing task is already active for this dataset. Wait for completion or abort it.");
             }
 
@@ -331,8 +331,8 @@ public class SpatialIndexLib {
     public static BasicTask scheduleOnceCleanTask(DatasetGraph dsg, TaskListener<BasicTask> taskListener) {
         Context cxt = dsg.getContext();
         BasicTask task = cxt.compute(SpatialIndexConstants.symSpatialIndexTask, (key, priorTaskObj) -> {
-            BasicTask priorTask = (BasicTask) priorTaskObj;
-            if (priorTask != null && !priorTask.isTerminated()) {
+            BasicTask priorTask = (BasicTask)priorTaskObj;
+            if (priorTask != null && !priorTask.hasTerminated()) {
                 throw new RuntimeException("A spatial indexing task is already active for this dataset. Wait for completion or abort it.");
             }
 

--- a/jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/task/BasicTask.java
+++ b/jena-geosparql/src/main/java/org/apache/jena/geosparql/spatial/task/BasicTask.java
@@ -53,7 +53,12 @@ public interface BasicTask extends Abortable {
     // XXX this might be different from whether the task actually transitioned into aborting state.
     boolean isAborting();
 
-    default boolean isTerminated() {
+    /**
+     * Whether the task has terminated.
+     *
+     * @implNote The original name {@code isTerminated} collided with the package−private method in {@link Thread}.
+     */
+    default boolean hasTerminated() {
         TaskState state = getTaskState();
         return TaskState.TERMINATED.equals(state);
     }

